File: src/main/java/gr/apt/persistence/dto/CreateNotificationDto.java
package gr.apt.persistence.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.math.BigInteger;
import java.util.Set;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreateNotificationDto {
    private String content;
    private BigInteger recipientPersonId;
    private Set<BigInteger> recipientRoleIds;

}

File: src/main/java/gr/apt/dto/notification/CreateNotificationDto.kt
package gr.apt.dto.notification

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import java.math.BigInteger

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class CreateNotificationDto(
    var content: String? = null,
    var recipientPersonId: BigInteger? = null,
    var recipientRoleIds: Set<BigInteger>? = null
)